src/node_file.cc

```diff
@@ -586,6 +586,37 @@ void AfterOpenFileHandle(uv_fs_t* req) {
   }
 }
 
+// Reverse the logic applied by path.toNamespacedPath() to create a
+// namespace-prefixed path.
+void FromNamespacedPath(std::string* path) {
+  replace_substring(path,  "\\\\?\\", "");
+  replace_substring(path,  "\\\\?\\UNC\\", "\\\\");
```

Comment:
This should only replace at the beginning of the string, right?

Reply:
Good point, do you think it's worth the potential performance hit to switch to a regex rooted on `^`?

Reply:
I don’t think a regexp is needed here, e.g.:

```cpp
void FromNamespacedPath(std::string* path) {
#ifdef _WIN32
  if (path->compare(0, 4, "\\\\?\\", 4) == 0) 
    *path = path->substr(4)
  else if (path.compare(0, 8, "\\\\?\\UNC\\",8) == 0)
    *path = "\\\\" + path.substr(8);
#endif
}
```